x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/license/DeleteLicenseResponse.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.protocol.xpack.license;
+
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+
+public class DeleteLicenseResponse extends AcknowledgedResponse {
+
+    private static final ConstructingObjectParser<DeleteLicenseResponse, Void> PARSER = new ConstructingObjectParser<>(
+        "delete_license_response", true, (a, v) -> new DeleteLicenseResponse((Boolean) a[0]));
+
+    static {
+        PARSER.declareBoolean(constructorArg(), AcknowledgedResponse.ACKNOWLEDGED);
+    }
+
+    public DeleteLicenseResponse() {
+    }
+
+    public DeleteLicenseResponse(boolean acknowledged) {
+        super(acknowledged);
+    }
+
+    public static DeleteLicenseResponse fromXContent(XContentParser parser) throws IOException {
+        return PARSER.parse(parser, null);
```

Comment:
there is a `parseAcknowledged(parser)` in `AcknowledgedResponse` that will let you remove all of the `PARSER` cruft here. I suspect it is also why `AcknowledgedResponse.ACKNOWLEDGED` is private.

Reply:
great. fixed in de96724